pkg/distro/fedora/images.go

```diff
@@ -200,40 +198,27 @@ func osCustomizations(
 			datastream = *imageConfig.DefaultOSCAPDatastream
 		}
 
-		oscapStageOptions := osbuild.OscapConfig{
-			Datastream:  datastream,
-			ProfileID:   oscapConfig.ProfileID,
-			Compression: true,
+		remediationConfig := oscap.RemediationConfig{
+			Datastream:         datastream,
+			ProfileID:          oscapConfig.ProfileID,
+			CompressionEnabled: true,
 		}
 
+		var tailoringConfig *oscap.TailoringConfig
 		if oscapConfig.Tailoring != nil {
-			newProfile, tailoringFilepath, tailoringDir, err := oscap.GetTailoringFile(oscapConfig.ProfileID)
-			if err != nil {
-				panic(fmt.Sprintf("unexpected error creating tailoring file options: %v", err))
-			}
-
-			tailoringOptions := osbuild.OscapAutotailorConfig{
-				NewProfile: newProfile,
-				Datastream: datastream,
-				ProfileID:  oscapConfig.ProfileID,
-				Selected:   oscapConfig.Tailoring.Selected,
-				Unselected: oscapConfig.Tailoring.Unselected,
+			tailoringConfig = &oscap.TailoringConfig{
+				RemediationConfig: remediationConfig,
+				Filepath:          filepath.Join(oscapDataDir, "tailoring.xml"),
+				NewProfile:        fmt.Sprintf("%s_osbuild_tailoring", oscapConfig.ProfileID),
+				Selected:          oscapConfig.Tailoring.Selected,
+				Unselected:        oscapConfig.Tailoring.Unselected,
 			}
-
-			osc.OpenSCAPTailorConfig = osbuild.NewOscapAutotailorStageOptions(
-				tailoringFilepath,
-				tailoringOptions,
-			)
-
-			// overwrite the profile id with the new tailoring id
-			oscapStageOptions.ProfileID = newProfile
-			oscapStageOptions.Tailoring = tailoringFilepath
-
-			// add the parent directory for the tailoring file
-			osc.Directories = append(osc.Directories, tailoringDir)
+			remediationConfig.ProfileID = tailoringConfig.NewProfile
+			remediationConfig.TailoringPath = tailoringConfig.Filepath
 		}
 
-		osc.OpenSCAPConfig = osbuild.NewOscapRemediationStageOptions(oscapDataDir, oscapStageOptions)
+		osc.OpenSCAPTailorConfig = osbuild.NewOscapAutotailorStageOptions(tailoringConfig)
+		osc.OpenSCAPConfig = osbuild.NewOscapRemediationStageOptions(oscapDataDir, &remediationConfig)
```

Comment:
I think that it would be nice to make `osc.OpenSCAPTailorConfig` and `osc.OpenSCAPConfig` be of their respective types from the `oscap` package, instead of `osbuild` stage options. This was the grand plan since the beginning for `OSCustomizations`, but we never finished it. Since you are already refactoring a lot of the code, it may make sense to make this change as well...

Reply:
Ah cool. Yeah this makes a lot of sense. I'll go ahead and do that